parser/src/lib.rs

```diff
@@ -241,7 +241,7 @@ impl Parser {
     fn factor(&mut self) -> Result<Expr> {
         let mut expr = self.unary()?;
 
-        if self.match_any_token(vec![TokenType::Slash, TokenType::Star]) {
+        if self.match_any_token(vec![TokenType::Slash, TokenType::Star, TokenType::Percent]) {
```

Comment:
Change this line to `if self.match_any_token(vec![TokenType::Percent, TokenType::Slash, TokenType::Star]) {`